arangod/Aql/EngineInfoContainerDBServerServerBased.cpp

```diff
@@ -199,7 +199,9 @@ EngineInfoContainerDBServerServerBased::buildSetupRequest(
   network::Headers headers;
   ClusterTrxMethods::addAQLTransactionHeader(trx, server, headers);
 
-  TRI_ASSERT(infoSlice.isObject() && infoSlice.get("clusterQueryId").isUInt());
+  TRI_ASSERT(infoSlice.isObject());
+  TRI_ASSERT(infoSlice.get("clusterQueryId").isNumber<QueryId>())
+      << "unexpected clusterQueryId: " << infoSlice.toJson();
```

Comment:
Can you explain for me what the difference between toJson and toString by semantic
I mean when I want to call toJson and when toString

Reply:
`toJson()` is the preferred way of doing this, because you will always get a full JSON representation of the value.
`toString()` returns the value as an std::string if the Slice is a string, and otherwise returns a pretty-printed JSON representation.